src/cli/self_update.rs

```diff
@@ -224,7 +224,9 @@ fn canonical_cargo_home() -> Result<String> {
         if cfg!(unix) {
             path_str = String::from("$HOME/.cargo");
         } else {
-            path_str = String::from(r"%USERPROFILE%\.cargo");
+            // Since this string serves as documentation, backslashes must be
+            // escaped in order to be displayed correctly on Windows.
+            path_str = String::from(r"%USERPROFILE%\\.cargo");
```

Comment:
I think rather than "serves as documentation" a comment along the lines of:

```rust
// Since this string will be interpolated into markdown which is then rendered
// for the user, backslashes must be escaped in order to make it through
// the markdown processing
``

Reply:
If you search for uses of this function (`canonical_cargo_home`). Not only its result passed
to `md` function but to raw stdout which doesn't need to be escaped.

Reply:
I see the function used in three places:

* https://github.com/rust-lang/rustup/blob/master/src/cli/self_update.rs#L319 -- where it is used to interpolate into a string which is then processed via the `md()` function
* https://github.com/rust-lang/rustup/blob/master/src/cli/self_update.rs#L817 -- where it is used similarly
* https://github.com/rust-lang/rustup/blob/master/src/cli/self_update.rs#L1155 -- where it is used as an interpolation into a shell string which is then only used under unix and thus irrelevant to this.

Could you please point out what I've missed?

Reply:
> If you search for uses of this function (`canonical_cargo_home`). Not only its result passed
> to `md` function but to raw stdout which doesn't need to be escaped.

Its true that markdown doesn't need its slashes escaped. Markdown with render both a single backslash and double backslash correctly. However, a single backslash to stdout wasn't rendering in windows command prompt.

> I think rather than "serves as documentation" a comment along the lines of

I do agree that I need to change the documentation to something closer to what @kinnison wrote. I'll wait for the discussion amongst collaborators to finish before adding another commit.

Let me know what you guys think. I'm totally fine with amending my PR in any way you see fit.

Reply:
Interesting, so you're saying that the single backslash makes it through the markdown okay, but that the Windows console output code somehow treats it as an escape?  If so, perhaps we ought to add escaping into our output routines rather than here on the inputs to them?  If I've misunderstood then please could you clarify?

Reply:
@kinnison In order to answer your question, I was going to confirm this by writing directly to stdout and compare that to the output of what went into the md() function. I built the new changes and ran `rustup self uninstall`. I was hoping to run `rustup-init.exe` and see duplicate output, but... 

It turns out my new commit produces a double slash in the console! ([from this particular line](https://github.com/rust-lang/rustup/blob/d6366cba52b00f517fa7f10f375cb91d40e02dc9/src/cli/self_update.rs#L179 )). The reason why a double backslash appeared here but not in the install phase is that the uninstall phase uses \`\` around the paths. It is markdown that is treating the backslash as an escape, not the console. 

We have yet another question that emerges. Should we add \`\` around all paths instead of adding escaping slashes?

Reply:
I don't mind either way.  In markdown documents I'd expect to see paths with ``s around them simply because I'd want them rendered monospace -- that's less of an issue with the console.  However for consistency perhaps we should ensure that all our console markdown is using ``s for paths, that way things ought to be easier to keep right.

Reply:
@lzutao thoughts on the above?